test/ngResource/resourceSpec.js

```diff
@@ -1356,5 +1357,36 @@ describe('resource', function() {
       );
   });
 
+  it('should cancel the request if timeout promise is resolved', function() {
+    var canceler = $q.defer();
+
+    $httpBackend.when('GET', '/CreditCard').respond({data: '123'});
+
+    var CreditCard = $resource('/CreditCard', {}, {
+      query: {
+        method: 'GET',
+        timeout: canceler.promise
+      }
+    });
+
+    CreditCard.query();
+
+    canceler.resolve();
+    expect(function() { $httpBackend.flush();}).toThrow(new Error("No pending request to flush !"));
```

Comment:
Nit: `expect($httpBackend.flush).toThrow(...);` is more concise and readable (here and below).
(Else you need a space after `flush();`.)